mountpoint-s3/src/prefetch.rs

```diff
@@ -76,22 +71,28 @@ pub struct Prefetcher<Client, Runtime> {
     inner: Arc<PrefetcherInner<Client, Runtime>>,
 }
 
-#[derive(Debug)]
 struct PrefetcherInner<Client, Runtime> {
-    client: Arc<Client>,
+    part_feed: Arc<dyn ObjectPartFeed<Client> + Send + Sync>,
```

Comment:
Any thoughts on whether this should be a generic type rather than dynamic dispatch? I guess we can get away with it here (unlike `Client`) because `ObjectPartFeed` is object-safe.

Reply:
It is a pragmatic choice which we can revert later on. The downside of a new generic type here is that it would propagate up to the root `S3FuseFilesystem` and end up touching most of the code in this crate. 
In fact, I had started working on an alternative option where I replaced `ObjectClient` with a new `ObjectStore` trait, with `get_object_parts` instead of `get_object` (and allowing for further divergence in the future), but that was a much larger change and would have slowed down work on the caching layer.
I still think something like that would be a preferable solution in the long term, but we can review the approach while we work on the cache or soon after.